SmilesMiles/View/TaskListView.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  TaskListView.swift
+//  SmilesMiles
+//
+//  Created by Pavan Sai Nallagoni on 2/15/23.
+//
+
+import SwiftUI
+
+struct TaskListView: View {
+    @State private var tasks: [Task] = []
+    @State private var newTask = ""
+
+    var body: some View {
+        VStack {
+            HStack {
+                TextField("New Task", text: $newTask)
+                Button(action: addTask) {
+                    Text("Add")
+                }
+            }
+            .padding()
+
+            List {
+                ForEach(tasks.sorted(by: { !$0.isCompleted && $1.isCompleted })) { task in
+                    HStack {
+                        Button(action: {
+                            if let index = tasks.firstIndex(of: task) {
+                                tasks[index].isCompleted.toggle()
+                                tasks.sort(by: { !$0.isCompleted && $1.isCompleted })
```

Comment:
What's the indent here?

Reply:
*Intent*, not indent 😅 .

Also, this could be DRYer.  On line 24, you have:
```
tasks.sorted(by: { !$0.isCompleted && $1.isCompleted })
```

and then here, on line 29, you have:
```
 tasks.sort(by: { !$0.isCompleted && $1.isCompleted }
 ```
 
 Is there a way to not repeat yourself and centralize this logic somewhere?  Perhaps a view model is a good place for it?

Reply:
True. I can make these changes after creating a view model.